src/Module/Search/Acl.php

```diff
@@ -0,0 +1,374 @@
+<?php
+
+namespace Friendica\Module\Search;
+
+use Friendica\BaseModule;
+use Friendica\Content\Widget;
+use Friendica\Core\Config;
+use Friendica\Core\Hook;
+use Friendica\Core\L10n;
+use Friendica\Core\Logger;
+use Friendica\Core\Protocol;
+use Friendica\Core\Session;
+use Friendica\Database\DBA;
+use Friendica\Model\Contact;
+use Friendica\Model\GContact;
+use Friendica\Model\Item;
+use Friendica\Network\HTTPException;
+use Friendica\Util\Network;
+use Friendica\Util\Proxy as ProxyUtils;
+use Friendica\Util\Strings;
+
+/**
+ * ACL selector json backend
+ *
+ * @package Friendica\Module\Search
+ */
+class Acl extends BaseModule
+{
+	public static function rawContent()
+	{
+		if (!local_user()) {
+			throw new HTTPException\UnauthorizedException(L10n::t('You must be logged in to use this module.'));
+		}
+
+		$start   = $_REQUEST['start']        ?? 0;
+		$count   = $_REQUEST['count']        ?? 100;
+		$search  = $_REQUEST['search']       ?? '';
+		$type    = $_REQUEST['type']         ?? '';
+		$conv_id = $_REQUEST['conversation'] ?? null;
+
+		// For use with jquery.textcomplete for private mail completion
+		if (!empty($_REQUEST['query'])) {
+			if (!$type) {
+				$type = 'm';
+			}
+			$search = $_REQUEST['query'];
+		}
+
+		Logger::info('ACL {action} - {subaction}', ['module' => 'acl', 'action' => 'content', 'subaction' => 'search', 'search' => $search, 'type' => $type, 'conversation' => $conv_id]);
+
+		$sql_extra = '';
+		$sql_extra2 = '';
+
+		if ($search != '') {
+			$sql_extra = "AND `name` LIKE '%%" . DBA::escape($search) . "%%'";
+			$sql_extra2 = "AND (`attag` LIKE '%%" . DBA::escape($search) . "%%' OR `name` LIKE '%%" . DBA::escape($search) . "%%' OR `nick` LIKE '%%" . DBA::escape($search) . "%%')";
+		}
+
+		// count groups and contacts
+		$group_count = 0;
+		if ($type == '' || $type == 'g') {
+			$r = q("SELECT COUNT(*) AS g FROM `group` WHERE NOT `deleted` AND `uid` = %d $sql_extra",
+				intval(local_user())
+			);
+			$group_count = (int) $r[0]['g'];
+		}
+
+		$sql_extra2 .= ' ' . Widget::unavailableNetworks();
+
+		$contact_count = 0;
+		if ($type == '' || $type == 'c') {
+			// autocomplete for editor mentions
+			$r = q("SELECT COUNT(*) AS c FROM `contact`
```

Comment:
All these Count queries are possible to transfer to `DBA::count()`, aren't they?

Reply:
Nope, because of the conditional `$sql_extra` and `$sql_extra2`. It is possible to transfer to `DBA` methods but it isn't straightforward.